visualization/app/codeCharta/util/streetLayoutGenerator.ts

```diff
@@ -0,0 +1,105 @@
+import { CodeMapNode, MetricData, Node, State, BlacklistType, TreeMapAlgorithm, LayoutAlgorithm } from "../codeCharta.model"
+import BoundingBox from "./algorithm/streetLayout/boundingBox"
+import VerticalStreet from "./algorithm/streetLayout/verticalStreet"
+import HorizontalStreet from "./algorithm/streetLayout/horizontalStreet"
+import House from "./algorithm/streetLayout/house"
+import Rectangle from "./algorithm/rectangle"
+import Point from "./algorithm/point"
+import { CodeMapHelper } from "./codeMapHelper"
+import TreeMap from "./algorithm/treeMap/treeMap"
+import SquarifiedTreeMap from "./algorithm/treeMap/squarifiedTreeMap"
+import { StreetOrientation } from "./algorithm/streetLayout/street"
+import { LayoutHelper, LayoutNode } from "./layoutHelper"
+
+export interface StreetLayoutValuedCodeMapNode {
+	data: CodeMapNode
+	value: number
+	rect: Rectangle
+	zOffset: number
+}
```

Comment:
Lets remove this interface and use the other `LayoutNode` which is the same, since we refactored some code

Reply:
After having a deeper look at the code, lets:
- keep the interface, but rename it to `StreetLayoutNode` (its still good enough isn't it?)
- remove its properties and extend the `LayoutNode` (yes, those interface are then the same for now, but make future additions very easy to apply)

That means, we can have different interfaces for each individual layoutAlgorithm, but when building the `Node[]` inside the LayoutHelper, we always cast/map those to a `LayoutNode`